#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_syllables.dpatch by  <A.Kral@sh.cvut.cz>
## 
## Written by Olivier Robert <poutifar@gmx.com>
##
## DP: This patch implements naming of note by syllables. 
## DP: Bug #556449

@DPATCH@
diff -urNad gtkguitune-0.8~/guitune.cc gtkguitune-0.8/guitune.cc
--- gtkguitune-0.8~/guitune.cc	2009-12-23 08:43:02.000000000 +0100
+++ gtkguitune-0.8/guitune.cc	2009-12-23 08:45:08.500730683 +0100
@@ -108,8 +108,14 @@
 {
     logview->setScale(LogView::german_scale_alt);
 }
-
-
+void MainWidget::setScaleSY()
+{
+    logview->setScale(LogView::syllables_scale);
+}
+void MainWidget::setScaleSYAlt()
+{
+    logview->setScale(LogView::syllables_scale_alt);
+}
 
 void MainWidget::showLogView()
 {
diff -urNad gtkguitune-0.8~/guitune.h gtkguitune-0.8/guitune.h
--- gtkguitune-0.8~/guitune.h	2008-09-16 11:56:09.000000000 +0200
+++ gtkguitune-0.8/guitune.h	2009-12-23 08:45:08.500730683 +0100
@@ -57,6 +57,8 @@
    void setScaleUSAlt();
    void setScaleGE();
    void setScaleGEAlt();
+   void setScaleSY();
+   void setScaleSYAlt();
    void showOszi();
    void hideOszi();
    void showLogView();
diff -urNad gtkguitune-0.8~/logview.cc gtkguitune-0.8/logview.cc
--- gtkguitune-0.8~/logview.cc	2008-09-16 11:56:09.000000000 +0200
+++ gtkguitune-0.8/logview.cc	2009-12-23 08:45:08.500730683 +0100
@@ -33,8 +33,10 @@
 
 const char *note_ger[12] = {"A","B", "H","C","Db","D","Eb","E","F","Gb","G","Ab"};
 const char *note_us[12] =  {"A","Bb","B","C","Db","D","Eb","E","F","Gb","G","Ab"};
+const char *note_sy[12] =  {"LA","SIb","SI","DO","REb","RE","MIb","MI","FA","SOLb","SOL","LAb"};
 const char *note_ger_alt[12] = {"A","B", "H","C","C#","D","D#","E","F","F#","G","G#"};
 const char *note_us_alt[12] =  {"A","A#","B","C","C#","D","D#","E","F","F#","G","G#"};
+const char *note_sy_alt[12] =  {"LA","LA#","SI","DO","DO#","RE","RE#","MI","FA","FA#","SOL","SOL#"};
 
 const char **note;
 
@@ -142,12 +144,18 @@
     case german_scale:
         note=note_ger;
         break;
+    case syllables_scale:
+        note=note_sy;
+        break;
     case us_scale_alt:
         note=note_us_alt;
         break;
     case german_scale_alt:
         note=note_ger_alt;
         break;
+    case syllables_scale_alt:
+        note=note_sy_alt;
+        break;
     }
     invalidate();
 }
diff -urNad gtkguitune-0.8~/logview.h gtkguitune-0.8/logview.h
--- gtkguitune-0.8~/logview.h	2008-09-16 11:56:09.000000000 +0200
+++ gtkguitune-0.8/logview.h	2009-12-23 08:45:08.500730683 +0100
@@ -36,7 +36,14 @@
     LogView();
     ~LogView();
 
-    enum {us_scale,us_scale_alt,german_scale,german_scale_alt};
+    enum {
+          us_scale,
+          us_scale_alt,
+          german_scale,
+          german_scale_alt,
+          syllables_scale,
+          syllables_scale_alt
+         };
 
     int  lfreq_pos(double lfreq);
     int  pos_note(int pos);
diff -urNad gtkguitune-0.8~/main.cc gtkguitune-0.8/main.cc
--- gtkguitune-0.8~/main.cc	2009-12-23 08:43:02.000000000 +0100
+++ gtkguitune-0.8/main.cc	2009-12-23 08:45:08.500730683 +0100
@@ -330,7 +330,10 @@
                    sigc::mem_fun(&m_wid, &MainWidget::setScaleGE) ));
         list_scale.push_back(RadioMenuElem( gr1,"G_erman-Alt",
                    sigc::mem_fun(&m_wid, &MainWidget::setScaleGEAlt) ));
-
+        list_scale.push_back(RadioMenuElem( gr1,"_Syllables",
+                   sigc::mem_fun(&m_wid, &MainWidget::setScaleSY) ));
+        list_scale.push_back(RadioMenuElem( gr1,"S_yllables-Alt",
+                   sigc::mem_fun(&m_wid, &MainWidget::setScaleSYAlt) ));
 
         // submenu Input Channel
         menu_input_rec = manage( new Menu() );
